worker/uniter/resolver.go

```diff
@@ -52,12 +54,15 @@ func (s *uniterResolver) NextOp(
 		return nil, resolver.ErrTerminate
 	}
 
-	// If the unit has completed a pre-series-upgrade hook (as noted by its
-	// state) then the uniter should idle in the face of all remote state
-	// changes accept for those which indicate termination - the unit is
-	// waiting to be shutdown.
-	if remoteState.UpgradeSeriesStatus == model.UpgradeSeriesPrepareCompleted {
-		return nil, resolver.ErrNoOperation
+	// Operations for series-upgrade need to be resolved early,
+	// in particular because no other operations should be run when the unit
+	// has completed preparation and is waiting for upgrade completion.
+	op, err := s.config.UpgradeSeries.NextOp(localState, remoteState, opFactory)
```

Comment:
Reading comments at line 277 raise a question for me:  Do we know this won't impact CAAS models?  Should we be paranoid and ensure it doesn't?  I think there are a few checks in state now, but not sure how complete they are.